src/ipld.rs

```diff
@@ -0,0 +1,243 @@
+use alloc::vec::Vec;
+use core::convert::TryInto;
+use core::num::TryFromIntError;
+
+use num_bigint::BigUint;
+
+use multihash::Code;
+use multihash::MultihashDigest;
+
+use libipld::cbor::DagCborCodec;
+use libipld::codec::Codec;
+use libipld::Cid;
+use libipld::Ipld;
+
+use crate::field::LurkField;
+
+pub const DAGCBOR: u64 = 0x71;
+
+pub trait IpldEmbed: Sized {
+    fn to_ipld(&self) -> Ipld;
+    fn from_ipld(ipld: &Ipld) -> Result<Self, IpldError>;
+}
+
+pub fn cbor_cid(code: u64, ipld: &Ipld) -> Cid {
+    Cid::new_v1(
+        code,
+        Code::Blake3_256.digest(DagCborCodec.encode(ipld).unwrap().as_ref()),
```

Comment:
We need it in order to be able to generate `Cid`s that current ipfs implementations know how to read (i.e. when we write the lurk ipfs client) but I will rename it and comment so that it's clear that this is for external use only